app/src/main/java/com/edu/news/screen/home/articles/ArticlesFragment.kt

```diff
@@ -0,0 +1,53 @@
+package com.edu.news.screen.home.articles
+
+import android.view.View
+import android.widget.Toast
+import com.edu.news.data.model.Article
+import com.edu.news.data.model.CategoryName
+import com.edu.news.data.source.ArticleRepository
+import com.edu.news.data.source.remote.ArticleRemoteDataSource
+import com.edu.news.data.source.remote.fetchData.GetDataFromUrl
+import com.edu.news.utils.base.BaseFragment
+import com.sun.news.R
+import kotlinx.android.synthetic.main.fragment_articles.view.*
+
+class ArticlesFragment private constructor(private val categoryName: String) : BaseFragment(),
+    ArticlesContract.View {
+
+    private val articlesAdapter = ArticlesAdapter()
+
+    override fun getLayoutResourceId() = R.layout.fragment_articles
+
+    override fun initView(view: View) {
+        view.recyclerViewArticles.adapter = articlesAdapter
+    }
+
+    override fun initData() {
+        val categoryType =
+            if (categoryName == CategoryName.TOP_HEADLINE.categoryName) "" else categoryName
+        ArticlePresenter(
+            ArticleRepository.getInstance(
+                ArticleRemoteDataSource.getInstance(
+                    GetDataFromUrl()
+                )
```

Comment:
thằng GetDataFromUrl() tạo luôn trong ArticleRemoteDataSource e
```suggestion
                ArticleRemoteDataSource.getInstance()
```